src/components/Filter/Filter.js

```diff
@@ -2,18 +2,20 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { InputField } from '@dhis2/ui-core'
 
-export const Filter = ({ text, onChange, onClear, placeholder }) => (
+export const Filter = ({ text, onChange, onClear, placeholder, type }) => (
     <InputField
         placeholder={placeholder}
         onChange={ref => (ref.value.length ? onChange(ref.value) : onClear())}
         value={text}
         dense
+        type={type}
     />
 )
 
 Filter.propTypes = {
     placeholder: PropTypes.string.isRequired,
     text: PropTypes.string.isRequired,
+    type: PropTypes.string.isRequired,
```

Comment:
Would it make sense to just let it fall back to the html default (text) if not provided instead of making it required?

Reply:
Yes, I believe `ui-core` does this already (since we didn't pass `type` before. But we can do it too. Anyhow, it shouldn't be `isRequired`. Fixed